client/web/src/components/views/subnet-router-view.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useMemo, useState } from "react"
+import { ReactComponent as CheckCircle } from "src/assets/icons/check-circle.svg"
+import { ReactComponent as Clock } from "src/assets/icons/clock.svg"
+import { ReactComponent as Plus } from "src/assets/icons/plus.svg"
+import { NodeData, NodeUpdaters } from "src/hooks/node-data"
+import Button from "src/ui/button"
+import Input from "src/ui/input"
+
+export default function SubnetRouterView({
+  readonly,
+  node,
+  nodeUpdaters,
+}: {
+  readonly: boolean
+  node: NodeData
+  nodeUpdaters: NodeUpdaters
+}) {
+  const advertisedRoutes = useMemo(
+    () => node.AdvertisingRoutes || [],
+    [node.AdvertisingRoutes]
+  )
+  const [inputOpen, setInputOpen] = useState<boolean>(
+    advertisedRoutes.length === 0
+  )
+  const [inputText, setInputText] = useState<string>("")
+
+  return (
+    <>
+      <h1 className="mb-1">Subnet router</h1>
+      <p className="description mb-5">
+        Add devices to your tailnet without installing Tailscale.{" "}
+        <a
+          href="https://tailscale.com/kb/1019/subnets/"
+          className="text-indigo-700"
+          target="_blank"
+        >
+          Learn more &rarr;
+        </a>
+      </p>
+      {inputOpen ? (
+        <div className="-mx-5 card shadow">
+          <p className="font-medium leading-snug mb-3">Advertise new routes</p>
+          <Input
```

Comment:
should we disable (or hide entirely) the input field when in readonly mode?

Reply:
oh good call